File: backend/schemas/detection.py
from datetime import datetime
from pydantic import BaseModel


class BoundingBox(BaseModel):
    x: float
    y: float
    width: float
    height: float


class DetectionResult(BaseModel):
    timestamp: datetime
    image_id: str
    model_type: str  # "thermal" | "rgb" | "fusion"
    classification: str  # "fire" | "no_fire"
    confidence: float
    severity: str  # "critical" | "high" | "medium" | "low"
    hotspots: list[BoundingBox]
    image_url: str

File: frontend/src/types/detection.ts
// types/detection.ts
// Shared detection types used by the live feed and WebSocket layer.
// The API-specific types (ModelResult, YoloResult, etc.) live in api/detection.ts.

export interface BoundingBox {
  x: number;
  y: number;
  width: number;
  height: number;
}

export interface DetectionResult {
  timestamp: string;
  image_id: string;
  model_type: string;
  classification: string;
  confidence: number;
  severity: string;
  hotspots: BoundingBox[];
  image_url: string;
}
